lib/Runtime/Executor/ThreadPoolExecutor.cpp

```diff
@@ -214,6 +215,12 @@ ThreadPoolExecutor::~ThreadPoolExecutor() {
 void ThreadPoolExecutor::run(const DAGNode *root,
                              std::unique_ptr<Context> context,
                              RunIdentifierTy runId, ResultCBTy cb) {
+  // Don't process new requests if the executor is shutting down.
+  if (shuttingDown_) {
+    cb(runId, ResultCode::Canceled, std::move(context));
```

Comment:
nit, should it be a ResultCode::Failed? It never really started.

Reply:
@gcatron I don't know what these mean, it would be good if they had some comments saying what each of them is. Canceled sounds more correct though just from guessing because there was nothing wrong with the request, it just got there too late.

Reply:
If you tell me what each of these are, I can add a comment in this PR to the ResultCode enum

Reply:
Yeah, they are a bit vague, canceled sounds good.